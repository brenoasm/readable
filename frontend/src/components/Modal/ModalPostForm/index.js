import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { compose } from 'redux';

import PostFormContainer from '../../../containers/PostFormContainer';
import Modal from '../../modal';
import WithForm from '../../forms/WithForm'
import PostForm from '../../forms/PostForm';
import Button from '../../buttons/Button';

const propTypes = {
  modalIsVisible: PropTypes.bool.isRequired,
  handleCancel: PropTypes.func,
  handleSubmit: PropTypes.func,
  disabledSubmit: PropTypes.bool
};

const defaultProps = {
  modalIsVisible: false,
  handleCancel: () => {},
  handleSubmit: () => {},
  disabledSubmit: false
};

const StyledModalPostForm = styled(Modal)`
  & > div {

      > div:first-child {
        min-height: 100px;
        margin: 5px;
        padding: 10px;
      }

      > div:last-child {
        border-top: 1px solid rgba(0, 0, 0, 0.15);
        height: 70px;
        padding: 10px;
        display: flex;
        justify-content: flex-end;

        button {
          margin: 5px 10px 0px 15px;
        }
      }
  }
`;

const ModalPostForm = ({ modalIsVisible, handleClose }) => (
  <PostFormContainer>
    <WithForm>
      <StyledModalPostForm show={modalIsVisible} handleClick={handleClose}>
        {({properties,
          handleInput,
          handleClearForm,
          handleSubmit,
          disabledSubmit}) => {

            const composedCancel = compose(
              handleClose,
              handleClearForm
            );

            const composedSubmit = compose(
              composedCancel,
              handleSubmit
            );

            return (
              <Fragment>
                <StyledModalPostForm.Body>
                  <PostForm properties={properties} handleInput={handleInput} />
                </StyledModalPostForm.Body>
                <StyledModalPostForm.Footer>
                  <Button
                    handleClick={composedCancel}
                    type="button"
                    text="Cancelar"
                  />
                  <Button
                    handleClick={composedSubmit}
                    type="button"
                    text="Concluir"
                    disabled={disabledSubmit}
                  />
                </StyledModalPostForm.Footer>
              </Fragment>
            )
        }}
      </StyledModalPostForm>
    </WithForm>
  </PostFormContainer>
);

ModalPostForm.propTypes = propTypes;
ModalPostForm.defaultProps = defaultProps;

export default ModalPostForm;
