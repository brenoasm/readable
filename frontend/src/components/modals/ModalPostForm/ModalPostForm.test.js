/* eslint-disable */

import React from 'react';

import ModalPostForm from './';

describe('ModalPostForm', () => {
  it('matches the snapshot', () => {
    const wrapper = shallow(<ModalPostForm />);
    expect(wrapper).toMatchSnapshot();
  });
});
