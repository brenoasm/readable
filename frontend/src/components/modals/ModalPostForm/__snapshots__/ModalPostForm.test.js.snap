// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalPostForm matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ModalPostForm
    disabledSubmit={false}
    handleCancel={[Function]}
    handleSubmit={[Function]}
    modalIsVisible={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <WithForm
        formProperties={Object {}}
        onPropertyChanges={[Function]}
        onSubmit={[Function]}
>
        <Styled(Modal)
                handleClick={undefined}
                show={false}
        >
                [Function]
        </Styled(Modal)>
</WithForm>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Styled(Modal)
          handleClick={undefined}
          show={false}
>
          [Function]
</Styled(Modal)>,
        "formProperties": Object {},
        "onPropertyChanges": [Function],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": [Function],
          "handleClick": undefined,
          "show": false,
        },
        "ref": null,
        "rendered": [Function],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithForm
          formProperties={Object {}}
          onPropertyChanges={[Function]}
          onSubmit={[Function]}
>
          <Styled(Modal)
                    handleClick={undefined}
                    show={false}
          >
                    [Function]
          </Styled(Modal)>
</WithForm>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Styled(Modal)
            handleClick={undefined}
            show={false}
>
            [Function]
</Styled(Modal)>,
          "formProperties": Object {},
          "onPropertyChanges": [Function],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": [Function],
            "handleClick": undefined,
            "show": false,
          },
          "ref": null,
          "rendered": [Function],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
